.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimeParser.cs

```diff
@@ -367,15 +365,24 @@ private DateTimeResolutionResult ParseUnspecificTimeOfDate(string text, DateObje
             var eod = this.config.UnspecificEndOfRegex.Match(text);
             if (eod.Success)
             {
-                ret.Timex = DateTimeFormatUtil.FormatDate(refDateTime) + "T23:59:59";
-                ret.FutureValue = refDateTime.Date.AddDays(1).AddSeconds(-1);
-                ret.PastValue = refDateTime.Date.AddDays(1).AddSeconds(-1);
-                ret.Success = true;              
+                ret = GetParsedResult(DateTimeFormatUtil.FormatDate(refDateTime), refDateTime, refDateTime);          
             }
 
             return ret;
         }
 
+        private DateTimeResolutionResult GetParsedResult(string timexPrefix, DateObject futureDate, DateObject pastDate)
```

Comment:
This should be renamed to make it clear what the method is doing. Maybe ResolveEndOfDay?
You can just pass pr and return what you need. All the parameters come from pr.